pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -331,6 +331,18 @@ type PipelineRunStatusFields struct {
 
 	// PipelineRunSpec contains the exact spec used to instantiate the run
 	PipelineSpec *PipelineSpec `json:"pipelineSpec,omitempty"`
+
+	// list of tasks that were skipped due to when expressions evaluating to false
+	// +optional
+	SkippedTasks []SkippedTask `json:"skippedTasks,omitempty"`
```

Comment:
No changes needed here. Looking at `json:...` here made me think what if someone tries to mock with the system by adding a bunch of `SkippedTasks` in the status. Looking at the overall `struct`, it seems possible to specify `Status` similar to `Spec` since most of the fields are marked as `optional` with `JSON` schema supported 🤔  is it something by design?